adafruit_midi/midi_message.py

```diff
@@ -82,57 +89,70 @@ class MIDIMessage:
     This is an *abstract* class.
     """
 
-    _STATUS = None
+    _STATUS: Optional[int] = None
     _STATUSMASK = None
-    LENGTH = None
+    LENGTH: Optional[int] = None
     CHANNELMASK = 0x0F
     ENDSTATUS = None
 
     # Commonly used exceptions to save memory
     @staticmethod
-    def _raise_valueerror_oor():
+    def _raise_valueerror_oor() -> None:
         raise ValueError("Out of range")
 
     # Each element is ((status, mask), class)
     # order is more specific masks first
-    _statusandmask_to_class = []
+    # Add better type hints for status, mask, class referenced above
+    _statusandmask_to_class: List[
+        Tuple[Tuple[Optional[bytes], Optional[int]], "MIDIMessage"]
+    ] = []
 
-    def __init__(self, *, channel=None):
+    def __init__(self, *, channel: Optional[int] = None) -> None:
         self._channel = channel  # dealing with pylint inadequacy
         self.channel = channel
 
     @property
-    def channel(self):
+    def channel(self) -> Optional[int]:
         """The channel number of the MIDI message where appropriate.
         This is *updated* by MIDI.send() method.
         """
         return self._channel
 
     @channel.setter
-    def channel(self, channel):
+    def channel(self, channel: int) -> None:
         if channel is not None and not 0 <= channel <= 15:
             raise ValueError("Channel must be 0-15 or None")
         self._channel = channel
 
     @classmethod
-    def register_message_type(cls):
+    def register_message_type(cls) -> None:
         """Register a new message by its status value and mask.
         This is called automagically at ``import`` time for each message.
         """
         ### These must be inserted with more specific masks first
         insert_idx = len(MIDIMessage._statusandmask_to_class)
         for idx, m_type in enumerate(MIDIMessage._statusandmask_to_class):
+            assert cls._STATUSMASK is not None
             if cls._STATUSMASK > m_type[0][1]:
                 insert_idx = idx
                 break
 
+        assert cls._STATUS is not None
+        assert cls._STATUSMASK is not None
         MIDIMessage._statusandmask_to_class.insert(
             insert_idx, ((cls._STATUS, cls._STATUSMASK), cls)
         )
 
     # pylint: disable=too-many-arguments
     @classmethod
-    def _search_eom_status(cls, buf, eom_status, msgstartidx, msgendidxplusone, endidx):
+    def _search_eom_status(
+        cls,
+        buf: Dict[int, bool],
+        eom_status: bool,
```

Comment:
I think `eom_status` might be  `Optional[int]` looking at usages in the code it seems to pass `ENDSTATUS` which has value of either None or an int in the different classes that define it.